JoinColony/block-ingestor
# Add formatting & linting

This PR adds a few components to the formatting/linting setup of block-ingestor to make it more consistent with the CDapp:
- Prettier config (with relevant ESLint plugin)
- lint-staged config responsible for linting and formatting 
- husky hooks to run lint-staged on commit and typecheck on push

## Testing
- Run `npm i`
- Try to commit a file with eslint error (it should either be fixed automatically or committing should fail)
- Try to commit a file with formatting issue (it should be formatted automatically)
- Try pushing to remote when there's a TypeScript error present (doesn't have to be committed; the push should fail)
